components/omega/doc/design/PGrad.md

```diff
@@ -0,0 +1,224 @@
+
+### 2.3 Requirement: Flexibility to support a high-order pressure gradient
+The centered pressure gradient will be insufficient for future versions of Omega that include ice shelf cavities and high resolution shelf breaks.
+The pressure gradient framework should be flexible enough to support a high-order pressure gradient in the future.
+The high order pressure gradient will be similar to [Adcroft et al. 2008](https://doi.org/10.1016/j.ocemod.2008.02.001).
+
+### 2.4 Requirement: Flexibility to support tidal forcing and sea level change
+In later versions of Omega, the pressure gradient will need to be able to include tidal forcing in the geopotential term.
+These tidal forcings include both the tidal potential and the self attraction and loading terms.
+Additionally, other long-term changes to the geoid can be included in the geopotential.
+
+### 2.5 Disired: Pressure gradient for barotropic mode
+
+For split barotropic-baroclinic timestepping, the pressure gradient should provide the bottom pressure gradient tendency in the barotropic mode.
+This will be added in a future version when split time stepping is implemented.
+
+### Desired:
+
+## 3 Algorithmic Formulation
+### 3.1 Centered Pressure Gradient
+In the layered non-Boussinesq [momentum equation](OmegaV1GoverningEqns.md#discrete-momentum) solved in Omega, the pressure gradient tendency term for edge $e$ and level $k$, $T^p_{e,k}$, includes the gradient of the pressure and the gradient of the geopotential,
+
+$$
+T^p_{e,k} = -\left[ \alpha_{i,k} \right]_e \nabla p_{i,k} - \nabla \Phi_{i,k},
+$$
+
+where the second term is necessary to account for tilted layers that occur when using a general vertical coordinate.
+In this equation, $\alpha_{i,k}$ is the specific volume for cell $i$ at the mid-point of level $k$, $p_{i,k}$ is the pressure, and $\Phi_{i,k}$ is the geopotential.
+The discrete gradient operator at an edge is:
+
+$$
+ \nabla {(\cdot)} = \frac{1}{d_e} \sum_{i\in CE(e)} -n_{e,i} (\cdot)_i
+$$
+
+where $d_e$ is the distance between cell centers, $CE(e)$ are the cells on edge $e$, and $n_{e,i}$ is the sign of the edge normal with respect to cell $i$.
+The horizontal averaging operator is:
+
+$$
+ [\cdot]_e = \frac{1}{2}\sum_{i\in CE(e)} (\cdot)_i
+$$
+
+Therefore, the centered pressure gradient will be calculated as:
+
+$$
+ T^p_{e,k} = \frac{1}{d_e} \left( [\alpha_{i,k}]_e \sum_{i \in CE(e)} n_{e,i} p_{k,i} + \sum_{i\in CE(e)} n_{e,i} \Phi_{k,i}\right),
+$$
+
+$$
+ = \frac{1}{d_e} \left(  \sum_{i \in CE(e)} n_{e,i} \left( [\alpha_{i,k}]_ep_{k,i} + \Phi_{k,i} \right) \right),
+$$
+
+### 3.2 Barotropic Pressure Gradient
+
+When split baroclinic-barotropic time stepping is implemented in the future, the barotropic pressure gradient will be calculated by the pressure gradient class.
+The barotropic pressure gradient is found by depth integrating the pressure gradient.
+The pressure is
+
+$$
+p(z) = p_b - g \int^z_{-h} \rho dz^\prime,
+$$
```

Comment:
Why is $z$ and not $\tilde{z}$ the coordinate of integration in this whole section?  And is it worth retaining this section or should we address it when we address the barotropic equation as a whole?

Reply:
I've decided to leave the barotropic pressure gradient to be discussed in the split timestepping design doc.